src/audio.rs

```diff
@@ -5,15 +5,30 @@ use luminance_sdl2::sdl2::audio::{AudioCallback, AudioDevice, AudioSpecDesired};
 
 pub const SAMPLE_RATE: i32 = 48000;
 
-/// A sound that is playing.
-struct AudioSource {
+macro_rules! impl_builder {
+    ( $( $field:ident : $type:ty ),* $(,)? ) => {
+        $(
+            pub fn $field(mut self, $field: $type) -> Self {
+                self.$field = $field;
+                self
+            }
+        )*
+    }
+}
+
+/// A sound that is playing or can be played.
+pub struct AudioSource {
     /// Which specific sample we're currently on.
```

Comment:
Indent maybe?

Reply:
Indent how? Line 21 is the documentation for line 22.